### ISSUE TITLE
[false-negative]SA_LOCAL_SELF_ASSIGNMENT can not detect the case like: this.foo = foo

### ISSUE THREAD

**Issue by wuchiuwong** [spotbugs/spotbugs#1145]
**Affects SpotBugs Version:** 
spotbugs-4.0.0-RC2

**Rule:**
SA_LOCAL_SELF_ASSIGNMENT

**Description:**
Unable to detect the case like: <code>this.foo = foo</code>
This rule is implemented in SpotBugs.jar\main\java\edu\umd\cs\findbugs\detect\FindLocalSelfAssignment2.java
The implementation of this rule is: first save the register operand of the load operation(<code>previousLoadOf = getRegisterOperand()</code>), 
and then compare whether the register operand of the store operation is the same as the former(<code>if (previousLoadOf == getRegisterOperand() && gotoCount < 2 && getPC() != previousGotoTarget)</code>), 
so it may not detect the self-assignment of class members.

**Code Sample demonstrating the issue:**
```
public BackgroundAndTitle()
{
	this.frame = frame;
}
```

**Expected outcome:**
false-negative

**Running SpotBugs through:**
command line

**Comment by Xiaoven** [comment]
`this.frame` is a field, so the pattern should be `SA_FIELD_SELF_ASSIGNMENT`.
